### ISSUE TITLE
Add cache node support to "big getter" functions

### ISSUE THREAD

**Issue by tinybike** [AugurProject/augur.js#96]
`getMarketInfo`, `getMarketsInfo`, `batchGetMarketInfo`, `getMarketPriceHistory`, and `getAccountTrades` should optionally be able to load data from [cache nodes](https://github.com/AugurProject/augur_node), verify it against the blockchain, and serve it to the UI.  If possible, the signatures for these methods should be unchanged -- to the UI, it's getting exactly the same data as before, just much more quickly!

(Tagging @kevinday in this issue b/c I suspect this will involve a fair bit of tweaking on the augur_node end as well...)

**Comment by kevinday** [comment]
Just wanted to comment to note that I'm starting to look at this. This will be a pretty significant increase in scope and responsibility for marketeer+cache nodes.

Just out of curiosity, how slow are the methods you mentioned? While I know they aren't fast for *all* markets at once, I thought we decided that they are acceptable for the 10 or so markets we will be loading at a time on a paginated basis. I'm not saying I don't think we need this, but I just want to make sure the cost and added complexity will be worth it.

**Comment by thinkloop** [comment]
The important one is `getMarketsInfo`. When you load the site currently you see the number of markets increasing by 10 every 1.5s or so. For thousands of markets this takes a minute or two for the site to fully load and be fully searchable.

**Comment by kevinday** [comment]
`getMarketsInfo` (only) is currently supported: http://augurnode9.eastus.cloudapp.azure.com:8547/getMarkets
(I should probably rename this endpoint to getMarketsInfo to be consistent btw).

Adding the additional methods the @tinybike is asking for may be nice to have, but I'm trying to determine if the benefit will be large enough to justify the increased complexity.

**Comment by thinkloop** [comment]
I believe this issue is about integrating that function with augurjs so that the ui can load from cache without having to change anything in the ui. 

Right now the ui does `augur.getMarketsInfo()` but it is only loading from chain. Once this issue is complete, that function should load from cache if available. Additionally, it should check the cache integrity. So something like:

1. Ping X number of cache nodes randomly (to avoid ddos) for one that is up
2. Load market data from it
3. Verify market hash for integrity
4. Return results hopefully maintaining exact same api

**Comment by kevinday** [comment]
Integrating cache nodes with augur.js is a separate matter that we are going to address soon. This issue is only discussing what the apis of a cache node should be, not how its integration will work once these apis are available.

**Comment by kevinday** [comment]
Cache nodes now support `getMarketInfo`,`getMarketsInfo`,  and `batchGetMarketInfo`. I'm doing `getMarketPriceHistory`, and `getAccountTrades` next.

**Comment by kevinday** [comment]
Cache nodes now have a getMarketPriceHistory endpoint available.

**Comment by kevinday** [comment]
These methods have been available for a few weeks now. See https://github.com/AugurProject/augur_node for usage. Closing
